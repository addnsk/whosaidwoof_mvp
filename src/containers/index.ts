export * from "./Header";
export * from "./Theme";
export * from "./SelectList";
