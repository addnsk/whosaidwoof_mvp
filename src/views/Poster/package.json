{
  "name": "poster-view",
  "version": "0.1.0",
  "private": true,
  "main": "Poster.tsx"
}
