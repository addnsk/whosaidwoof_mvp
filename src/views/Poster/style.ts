import styled from "styled-components";

export const Background = styled.section`
  width: 140px;
  height: 140px;
  border-radius: 49px;

  background-color: ${({ theme }) => theme.colors.accentLight};
  box-shadow: inset 0px 4px 4px rgba(0, 0, 0, 0.5);

  position: relative;

  & img {
    width: 189px;
    height: 149px;
    margin: 0;
    padding: 0;
    position: absolute;
    bottom: -1px;
    left: 0;
  }

  @media (min-height: 650px) {
    transform: scale(1.5);
  }

  @media (min-height: 900px) {
    transform: scale(1.9);
  }
`;
