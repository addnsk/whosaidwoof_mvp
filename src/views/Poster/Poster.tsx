import * as React from "react";
import img from "./img/dog.png";
import { Background } from "./style";

export const Poster: React.FC = (props): JSX.Element => {
  return (
    <Background {...props}>
      <img src={img} alt="Собака"></img>
    </Background>
  );
};
