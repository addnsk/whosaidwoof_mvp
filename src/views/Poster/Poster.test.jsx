import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { Theme } from "containers";
import { Poster } from "./Poster";

describe("Poster component", () => {
  it("Poster renders", () => {
    render(
      <Theme>
        <Poster data-testid="poster" />
      </Theme>
    );
    expect(screen.queryByTestId("poster")).toBeInTheDocument();
    expect(screen.getByRole("img")).toBeInTheDocument();
    expect(screen.getByAltText(/собака/i)).toBeInTheDocument();

    expect(screen.queryByTestId("poster")).toMatchSnapshot();
  });
});
