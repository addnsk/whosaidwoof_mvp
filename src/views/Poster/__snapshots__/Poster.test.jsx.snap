// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poster component Poster renders 1`] = `
<section
  class="sc-dlfnuX jALsTq"
  data-testid="poster"
>
  <img
    alt="Собака"
    src="dog.png"
  />
</section>
`;
